src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -454,6 +461,53 @@ public AgentSettings LoadSettings()
             }
         }
 
+#if OS_WINDOWS
+        private void ConfigureAutoLogonIfNeeded(CommandSettings command)
+        {
+            Trace.Entering();
+
+            if (!command.GetEnableAutoLogon())
+            {
+                return;
+            }
+
+            Trace.Info("Agent is going to run as process setting up the 'AutoLogon' capability for the agent.");
+            try
+            {                
+                var autoLogonConfigManager = HostContext.GetService<IAutoLogonConfigurationManager>();
+                autoLogonConfigManager.Configure(command);
+
+                if (autoLogonConfigManager.RestartNeeded())
```

Comment:
Even for the same user I think it makes sense to restart and let the agent launch via the configured changed. Otherwise any subtle difference between the two invocations becomes harder to root cause. Also if anything is hindering the auto-logon/startup settings then you find out immediately, rather than some time later after all is forgotten.